src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs

```diff
@@ -158,7 +158,7 @@ private static SafeFileHandle OpenHandle(string fullPath, bool asDirectory)
                 Interop.Kernel32.GenericOperations.GENERIC_WRITE,
                 FileShare.ReadWrite | FileShare.Delete,
                 FileMode.Open,
-                asDirectory ? Interop.Kernel32.FileOperations.FILE_FLAG_BACKUP_SEMANTICS : 0);
```

Comment:
My suggestion is that we focus this PR to only the OSX/Unix changes. Let's address any Windows concerns separately. Can you please revert the Windows changes?

Reply:
I'll open an issue after this PR goes through to address the Windows thing.
👍